### PR TITLE
[3.4] refactoring:  Customer controller

### PR DESCRIPTION
**Pull request by nickAS21**
## Pull Request description

Put your PR description here instead of this sentence.   

## General checklist

- [x] You have reviewed the guidelines [document](https://docs.google.com/document/d/1wqcOafLx5hth8SAg4dqV_LV3un3m5WYR8RdTJ4MbbUM/edit?usp=sharing).
- [x] PR name contains fix version. For example, "[3.3.4] Hotfix of some UI component" or "[3.4] New Super Feature".
- [x] Description references specific [issue](https://github.com/thingsboard/thingsboard/issues).
- [x] Description contains human-readable scope of changes.
- [x] Description contains brief notes about what needs to be added to the documentation.
- [x] No merge conflicts, commented blocks of code, code formatting issues.
- [x] Changes are backward compatible or upgrade script is provided.
- [x] Similar PR is opened for PE version to simplify merge. Required for internal contributors only.

## Back-End feature checklist

- [x] Added corresponding unit and/or integration test(s). Provide written explanation in the PR description if you have failed to add tests.
- [x] If new dependency was added: the dependency tree is checked for conflicts.
- [x] If new service was added: the service is marked with corresponding @TbCoreComponent, @TbRuleEngineComponent, @TbTransportComponent, etc.
- [x] If new REST API was added: the RestClient.java was updated, issue for [Python REST client](https://github.com/thingsboard/thingsboard-python-rest-client) is created.

### CODE REVIEW

File: application/src/main/java/org/thingsboard/server/service/entitiy/customer/DefaultTbCustomerService.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+import org.thingsboard.server.service.entitiy.AbstractTbEntityService;
+import org.thingsboard.server.service.security.model.SecurityUser;
+
+import java.util.List;
+
+@Service
+@TbCoreComponent
+@AllArgsConstructor
+public class DefaultTbCustomerService extends AbstractTbEntityService implements TbCustomerService {
+
```
Comment (YevhenBondarenko) [inline application/src/main/java/org/thingsboard/server/service/entitiy/customer/DefaultTbCustomerService.java:36]: remove extra line

File: application/src/main/java/org/thingsboard/server/service/entitiy/DefaultTbNotificationEntityService.java [hunk 2]
```diff
@@ -202,6 +203,21 @@ public void notifyDeleteAlarm(Alarm alarm, SecurityUser user, List<EdgeId> relat
         sendAlarmDeleteNotificationMsg(alarm, relatedEdgeIds);
     }
 
+    @Override
+    public void notifySaveCustomer(Customer customer, ActionType actionType, SecurityUser user, Object... additionalInfo) {
```
Comment (YevhenBondarenko) [inline application/src/main/java/org/thingsboard/server/service/entitiy/DefaultTbNotificationEntityService.java:207]: Name should be like other methods `notifyCreateOrUpdate`...
The asset method has the same logic, maybe should create method `notifyCreateOrUpdateEntity`?

File: application/src/main/java/org/thingsboard/server/service/entitiy/DefaultTbNotificationEntityService.java [hunk 3]
```diff
@@ -202,6 +203,21 @@ public void notifyDeleteAlarm(Alarm alarm, SecurityUser user, List<EdgeId> relat
...
+        logEntityAction(customer.getTenantId(), customer.getId(), customer, customer.getId(), actionType, user, additionalInfo);
+        if (customer.getId() != null) {
```
Comment (YevhenBondarenko) [inline application/src/main/java/org/thingsboard/server/service/entitiy/DefaultTbNotificationEntityService.java:209]: this condition always is true
see `notifyCreateOrUpdateAsset`

File: application/src/main/java/org/thingsboard/server/service/entitiy/customer/DefaultTbCustomerService.java (hunk 1 above)
Comment (nickAS21) [reply application/src/main/java/org/thingsboard/server/service/entitiy/customer/DefaultTbCustomerService.java:36]: OK

File: application/src/main/java/org/thingsboard/server/service/entitiy/DefaultTbNotificationEntityService.java (hunk 3 above)
Comment (nickAS21) [reply application/src/main/java/org/thingsboard/server/service/entitiy/DefaultTbNotificationEntityService.java:209]: OK

File: application/src/main/java/org/thingsboard/server/service/entitiy/DefaultTbNotificationEntityService.java (hunk 2 above)
Comment (nickAS21) [reply application/src/main/java/org/thingsboard/server/service/entitiy/DefaultTbNotificationEntityService.java:207]: Ok

File: application/src/main/java/org/thingsboard/server/service/entitiy/DefaultTbNotificationEntityService.java [hunk 4]
```diff
@@ -145,12 +145,10 @@ public void notifyAssignDeviceToTenant(TenantId tenantId, TenantId newTenantId,
     }
 
     @Override
-    public void notifyCreateOrUpdateAsset(TenantId tenantId, AssetId assetId, CustomerId customerId, Asset asset,
-                                          ActionType actionType, SecurityUser user, Object... additionalInfo) {
-        logEntityAction(tenantId, assetId, asset, customerId, actionType, user, additionalInfo);
-
-        if (actionType == ActionType.UPDATED) {
-            sendEntityNotificationMsg(asset.getTenantId(), asset.getId(), EdgeEventActionType.UPDATED);
+    public  <E extends HasName, I extends EntityId>  void notifyCreateOrUpdateEntity(TenantId tenantId, I entityId, E customer, CustomerId customerId, ActionType actionType, SecurityUser user, Object... additionalInfo) {
```
Comment (YevhenBondarenko) [inline application/src/main/java/org/thingsboard/server/service/entitiy/DefaultTbNotificationEntityService.java:148]: should be `E entity`

**Review by YevhenBondarenko — APPROVED** [review]
LGTM

**Pull request closed without merging**